### DISCUSSION TITLE
Switching windows according to current space

### DISCUSSION THREAD

**Discussion by 24unix** [asmvik/yabai#649] [Q&A] [answered]
Hi, I finally got most of the stuff running which I need.

As I had to split up spaces for each display, I ended up having 24 spaces on 3 monitors.

This works like before with mission control and shortcuts:

`ctrl - 1 : yabai -m space --focus 9; yabai -m space --focus 1; yabai -m space --focus 17
ctrl - 2 : yabai -m space --focus 10; yabai -m space --focus 2; yabai -m space --focus 18
ctrl - 3 : yabai -m space --focus 11; yabai -m space --focus 3; yabai -m space --focus 19
ctrl - 4 : yabai -m space --focus 12; yabai -m space --focus 4; yabai -m space --focus 20
ctrl - 5 : yabai -m space --focus 13; yabai -m space --focus 5; yabai -m space --focus 21
ctrl - 6 : yabai -m space --focus 14; yabai -m space --focus 6; yabai -m space --focus 22
ctrl - 7 : yabai -m space --focus 15; yabai -m space --focus 7; yabai -m space --focus 23
ctrl - 8 : yabai -m space --focus 16; yabai -m space --focus 8; yabai -m space --focus 24
`

Now I need some means of moving the windows. Before I was able to grab a window by the titlebar and hit eg. ctrl-3 to switch to my desktop #3 and take the window with it.

But, now I've got 24 spaces, but in deed 8 logical desktops.

So, when I've got a windows on my main desktop, (spaces 9, 1 & 17), on the left monitor (space 9), and I want it to got to my second Desktop, it must go to space 10, and so on.

Is there an easy way to make that possible for desktop 1-8 aka spaces 1-24?

**Comment by dominiklohmann** [comment, marked as answer]
Rewrite your keybindings like this:

```
ctrl - 1 : yabai -m space --focus $(yabai -m query --displays --display | jq -er '.spaces[0]')
ctrl - 2 : yabai -m space --focus $(yabai -m query --displays --display | jq -er '.spaces[1]')
```

**Reply by 24unix** [reply]
Thanks four your reply, but I found a solution without using skhd, see: https://github.com/koekeishiya/yabai/discussions/759

But I still tried your solution, it only switched one display.

**Reply by dominiklohmann** [reply]
The suggestion works for the current display. You can specify the display selector after `--display` in the query, e.g.,

```
# focus space 1 on display 2
ctrl - 1 : yabai -m space --focus $(yabai -m query --displays --display 2 | jq -er '.spaces[0]')
```

The default selector takes the current display.
